lib/blocs/location_provider_cubit.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:async';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:latlong/latlong.dart';
+import 'package:trufi_core/models/location_state.dart';
+
+class LocationProviderCubit extends Cubit<LocationState> {
+  LocationProviderCubit() : super(const LocationState());
+
+  Future<void> start() async {
+    final LocationPermission status = await Geolocator.checkPermission();
+    if (!(status == LocationPermission.always ||
+        status == LocationPermission.whileInUse)) {
+      return;
+    }
+
+    emit(state.copyWith(
+      locationStreamSubscription: Geolocator.getPositionStream(
+              desiredAccuracy: LocationAccuracy.high, distanceFilter: 10)
+          .listen((position) {
+        emit(state.copyWith(
+          currentLocation: LatLng(position.latitude, position.longitude),
+        ));
+      }),
+    ));
+  }
+
+  void stop() {
+    if (state.locationStreamSubscription != null) {
+      state.locationStreamSubscription.cancel();
+    }
+  }
+
+  // Getter
+
+  Future<LatLng> getCurrentLocation() async {
+    final LocationPermission status = await Geolocator.checkPermission();
+
+    if (status == LocationPermission.always ||
+        status == LocationPermission.whileInUse) {
+      final Position position = await Geolocator.getCurrentPosition(
+          desiredAccuracy: LocationAccuracy.high);
+
+      return position != null
+          ? LatLng(position.latitude, position.longitude)
+          : null;
```

Comment:
`getCurrentPosition()` will never return a null value, instead it will generate Throws:

> Throws a [TimeoutException] when no location is received within the supplied [timeLimit] duration. Throws a [PermissionDeniedException] when trying to request the device's location when the user denied access. Throws a [LocationServiceDisabledException] when the user allowed access, but the location services of the device are disabled.

for now I think is ok if we just add `.catchError((error) => null)` to ignore throws.

Reply:
The throws handling should be handle at the top level, but the error pop ups appears when the location is null, maybe it could part of the error- pop ups improvements

Reply:
I would not recommend swallowing throws. https://en.wikipedia.org/wiki/Error_hiding

Reply:
the problem is if `getCurrentPosition()` throws an error instead returning null, the error pop up will never appears

Reply:
Great catch :) thank you!

Reply:
yes

Reply:
#435